src/Microsoft.TestPlatform.Common/ExtensionFramework/Utilities/LazyExtension.cs

```diff
@@ -78,17 +68,12 @@ public LazyExtension(TestPluginInformation pluginInfo, Type metadataType)
         /// <param name="metadata">Test extension metadata</param>
         public LazyExtension(TestPluginInformation pluginInfo, TMetadata metadata)
         {
-            if (pluginInfo == null)
-            {
-                throw new ArgumentNullException("pluginInfo"); ;
-            }
-
             if (metadata == null)
```

Comment:
the above call is reduced to ?? but this one is not. use ?? for both (also to avoid changing the order of execution, which is imho very low risk here). Same below.